src/hippopt/base/optimization_solver.py

```diff
@@ -0,0 +1,48 @@
+import abc
+import dataclasses
+from typing import List, Tuple, TypeVar
+
+import casadi as cs
+
+from hippopt.base.optimization_object import TOptimizationObject
+
+TOptimizationSolver = TypeVar("TOptimizationSolver", bound="OptimizationSolver")
+
+
+@dataclasses.dataclass
+class OptimizationSolver(abc.ABC):
+    @abc.abstractmethod
+    def generate_optimization_objects(
+        self, input_structure: TOptimizationObject | List[TOptimizationObject], **kwargs
+    ) -> TOptimizationObject | List[TOptimizationObject]:
```

Comment:
That depends on the input. Maybe we can enforce it with the ``Generic`` thingy you mentioned yesterday?

Reply:
Yep, this seems a good use-case for that

Reply:
I have checked how to do it and I would like to implement it. Nonetheless, this would badly conflict with some code I started developing before receiving the review (https://github.com/ami-iit/hippopt/tree/ocStructure). I will tackle this in the next PR.

Reference issue: https://github.com/ami-iit/hippopt/issues/5